validator/main.go

```diff
@@ -675,35 +769,65 @@ func (d *Driver) validate() error {
 		}
 	}
 
-	statusFile := driverStatusFile
+	return d.createStatusFile(isHostDriver)
+}
+
+func (d *Driver) createStatusFile(isHostDriver bool) error {
+	var nvidiaDriverRoot, nvidiaDevRoot, driverRootCtrPath, devRootCtrPath string
 	if isHostDriver {
-		statusFile = hostDriverStatusFile
+		nvidiaDriverRoot = hostRootFlag
+		driverRootCtrPath = "/host"
+		nvidiaDevRoot = hostRootFlag
+		devRootCtrPath = "/host"
+	} else {
+		nvidiaDriverRoot = driverRootFlag
+		driverRootCtrPath = "/driver-root"
+		nvidiaDevRoot = devRootFlag
+
+		devRootCtrPath = driverRootCtrPath
+		if devRootFlag == hostRootFlag {
+			devRootCtrPath = "/host"
+		} else if devRootFlag != "" && devRootFlag != driverRootFlag {
+			// a custom dev root, which is not equal to host root or driver root,
+			// will be mounted in the operand container at /dev-root
+			devRootCtrPath = "/dev-root"
+		}
```

Comment:
If we decided to always mount the devPath to `/dev-root/dev` inside of our operand containers, then this code block could simplify to 
```
devRootCtrPath = /dev-root
```

Reply:
Agreed, it is simpler. I guess it doesn't matter what the mount path is as long as the path to be mounted remains customisable

Reply:
There are two aspects here. The first is giving operand containers access to GPUs -- as is done by the NVIDIA Container Toolkit. This will always inject the device nodes at `/dev` since this is where the driver expects them.

The second is for CDI spec generation which accepts an explicit dev root -- in which case a fixed path is fine.

Reply:
Yes, I was eluding to always mounting the device nodes to `/dev-root/dev` for the purpose of CDI spec generation.

Reply:
I have removed devPath from the ClusterPolicy API now. This code has been simplified now.